src/modules/objects/services/objects.service.ts

```diff
@@ -0,0 +1,328 @@
+import { randomUUID } from 'crypto';
+
+import { DataService, PlayerTicketService } from '@meemoo/admin-core-api';
+import { Injectable, Logger } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { Pagination } from '@studiohyperdrive/pagination';
+import got, { Got } from 'got';
+import { compact, find, get, intersection, set, unset } from 'lodash';
+
+import { Configuration } from '~config';
+
+import { ObjectsQueryDto, SearchFilter } from '../dto/objects.dto';
+import { QueryBuilder } from '../elasticsearch/queryBuilder';
+import {
+	ElasticsearchObject,
+	ElasticsearchResponse,
+	License,
+	MediaFormat,
+	ObjectMedia,
+	ObjectMediaWithAggregations,
+} from '../objects.types';
+
+import { SessionUserEntity } from '~modules/users/classes/session-user';
+import { VisitsService } from '~modules/visits/services/visits.service';
+
+@Injectable()
+export class ObjectsService {
+	private logger: Logger = new Logger(ObjectsService.name, { timestamp: true });
+	private gotInstance: Got;
+
+	constructor(
+		private configService: ConfigService<Configuration>,
+		protected dataService: DataService,
+		protected playerTicketService: PlayerTicketService,
+		private visitsService: VisitsService
+	) {
+		this.gotInstance = got.extend({
+			prefixUrl: this.configService.get('ELASTIC_SEARCH_URL'),
+			resolveBodyOnly: true,
+			responseType: 'json',
+		});
+	}
+
+	public async adaptESResponse(
+		esResponse: ElasticsearchResponse,
+		referer: string
+	): Promise<ElasticsearchResponse> {
+		// merge 'film' aggregations with 'video' if need be
+		if (esResponse.aggregations?.dcterms_format?.buckets) {
+			esResponse.aggregations.dcterms_format.buckets =
+				esResponse.aggregations.dcterms_format.buckets.filter((bucket) => {
+					if (bucket.key === 'film') {
+						const videoBucket = find(esResponse.aggregations.dcterms_format.buckets, {
+							key: 'video',
+						});
+						if (videoBucket) {
+							// there is also a video bucket: add film counts to this bucket
+							videoBucket.doc_count += bucket.doc_count;
+							return false; // filter out current film bucket
+						}
+						// there is no video bucket: rename the film bucket to video bucket
+						bucket.key = 'video';
+						return true; // include newly renamed video bucket in response
+					}
+					return true; // not a film bucket -> include in response
+				});
+		}
+
+		// sanity check
+		const nrHits = get(esResponse, 'hits.total.value');
+		if (!nrHits) {
+			return esResponse;
+		}
+
+		// there are hits - resolve thumbnail urls
+		esResponse.hits.hits = await Promise.all(
+			esResponse.hits.hits.map(async (hit) => {
+				hit._source.schema_thumbnail_url =
+					await this.playerTicketService.resolveThumbnailUrl(
+						hit._source.schema_thumbnail_url,
+						referer
+					);
+				return hit;
+			})
+		);
+
+		return esResponse;
```

Comment:
the other fields can also be simplified, so we don't expose elasticsearch format to the client:
See task:
```
  "aggregations": {...},
  "items": IeObject[],
  "total": number,
```